File: 02-04-2024_Isomorphic_Strings.java
public class Solution {
    public boolean isIsomorphic(String s, String t) {
        int[] S = new int[200];
        int[] T = new int[200];
        Arrays.fill(S, 0);
        Arrays.fill(T, 0);

        int len = s.length();
        if (len != t.length()) {
            return false;
        }
        for (int i = 0; i < len; i++) {
            if (S[s.charAt(i)] != T[t.charAt(i)]) {
                return false;
            }
            S[s.charAt(i)] = i + 1;
            T[t.charAt(i)] = i + 1;
        }
        return true;
    }
}

File: 02-04-2024_Isomorphic_Strings.py
class Solution:
    def isIsomorphic(self, s: str, t: str) -> bool:
        S = [0] * 200
        T = [0] * 200

        length = len(s)
        if length != len(t):
            return False
        for i in range(length):
            if S[ord(s[i])] != T[ord(t[i])]:
                return False
            S[ord(s[i])] = i + 1
            T[ord(t[i])] = i + 1
        return True
